GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace GenericsHomework
+{
+    public class Node<T> : ICollection<T>
+    {
+        private T _Value;
+        private Node<T> _Next;
+
+        public Node(T value)
+        {
+            _Value = value;
+            _Next = this;
+        }
+
+        public T Value
+        {
+            get
+            {
+                return _Value;
+            }
+
+            private set
+            {
+                _Value = value;
+            }
+
+        }
```

Comment:
This would be cleaner as an auto property.
```suggestion
        public T Value { get; private set; }
```